File: app/src/main/java/com/primoz/timetimer/realmObjects/Workout.java
/*
package com.primoz.timetimer.realmObjects;

import io.realm.Realm;
import io.realm.RealmList;
import io.realm.RealmModel;
import io.realm.RealmObject;

*/
/**
 * Created by Primož on 21/08/2017.
 *//*


public class Workout extends RealmObject {

    String name;
    int timeTotal;
    int timeWork;
    int timeRest;
    int timeRounds;

    public static void delete(Realm realm, String name) {
        Workout workout = realm.where(Workout.class).equalTo("name", name).findFirst();
        // Otherwise it has been deleted already.
        if (workout != null) {
            workout.deleteFromRealm();
        }
    }

    public static void createWorkout(Realm realm, String name, int timeTotal, int timeWork, int timeRest, int timeRounds) {
        WorkoutList list = realm.where(WorkoutList.class).findFirst();
        RealmList<Workout> workouts = list.getWorkoutRealmList();
        Workout workout = realm.createObject(Workout.class);
        workout.name = name;
        workout.timeTotal = timeTotal;
        workout.timeWork = timeWork;
        workout.timeRest = timeRest;
        workout.timeRounds = timeRounds;
        workouts.add(workout);
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setTimeTotal(int timeTotal) {
        this.timeTotal = timeTotal;
    }

    public void setTimeWork(int timeWork) {
        this.timeWork = timeWork;
    }

    public void setTimeRest(int timeRest) {
        this.timeRest = timeRest;
    }

    public void setTimeRounds(int timeRounds) {
        this.timeRounds = timeRounds;
    }

    public String getName() {
        return name;
    }

    public int getTimeTotal() {
        return timeTotal;
    }

    public int getTimeWork() {
        return timeWork;
    }

    public int getTimeRest() {
        return timeRest;
    }

    public int getTimeRounds() {
        return timeRounds;
    }

    public int getTotalSessionSeconds() {
        int trainingTogether = getTimeWork() * getTimeRounds();
        int restTogether = getTimeRest() * getTimeRounds();
        return trainingTogether + restTogether;
    }
}
*/

File: app/src/main/java/com/primoz/timetimer/data_mvp/Workout.kt
package com.primoz.timetimer.data_mvp

import io.realm.Realm
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.kotlin.createObject
import io.realm.kotlin.where

open class Workout : RealmObject() {

    @PrimaryKey
    var workoutID: Int = 0
    var name: String = ""
    var timeWork: Int = 0
    var timeRest: Int = 0
    var timeRounds: Int = 0
    val totalSessionSeconds: Int
        get() {
            val trainingTogether = timeWork * timeRounds
            val restTogether = timeRest * timeRounds
            return trainingTogether + restTogether
        }

    companion object {

        fun delete(realm: Realm, workoutID: Int) {
            val workout = realm.where<Workout>().equalTo("workoutID", workoutID).findFirst()
            // Otherwise it has been deleted already.
            workout?.deleteFromRealm()
        }

        fun createWorkout(realm: Realm, name: String, timeWork: Int, timeRest: Int, timeRounds: Int) {
            val list = realm.where<WorkoutList>().findFirst()
            val workouts = list!!.workoutRealmList

            var nextID = realm.where<Workout>().max("workoutID")
            nextID = if (nextID == null) 1 else nextID.toLong() + 1 //Gets MAX KEY_ID_PROGRAM +1 to be set

            val workout = realm.createObject<Workout>(nextID)
            workout.name = name
            workout.timeWork = timeWork
            workout.timeRest = timeRest
            workout.timeRounds = timeRounds
            workouts!!.add(workout)
        }

        fun exists(realm: Realm, workoutName: String): Boolean {
            val workout = realm.where<Workout>().equalTo("name", workoutName).findFirst()
            if(workout != null) return true
            return false
        }
    }
}
